cpp/src/arrow/io/file.cc

```diff
@@ -580,13 +580,21 @@ MemoryMappedFile::~MemoryMappedFile() {}
 
 Status MemoryMappedFile::Create(const std::string& path, int64_t size,
                                 std::shared_ptr<MemoryMappedFile>* out) {
+  int ret;
   std::shared_ptr<FileOutputStream> file;
   RETURN_NOT_OK(FileOutputStream::Open(path, &file));
 #ifdef _MSC_VER
-  _chsize_s(file->file_descriptor(), static_cast<size_t>(size));
+  ret = _chsize_s(file->file_descriptor(), static_cast<size_t>(size));
 #else
-  ftruncate(file->file_descriptor(), static_cast<size_t>(size));
+  ret = ftruncate(file->file_descriptor(), static_cast<size_t>(size));
 #endif
+  
+  if (ret != 0) {
+    std::stringstream ss;
+    ss << "Unable to create file, error: " << std::strerror(errno);
+    return Status::IOError(ss.str());
+  } 
```

Comment:
Can you use the `CheckOpenResult` function for this? Adding the errno to the error string generated there would be useful

Reply:
Ack. Perhaps I should rename "CheckOpenResult" to something more generic like "CheckFileOpsResult"?

Reply:
Sure, that works for me